controller/program/tests/tests/rebalance/chain.rs

```diff
@@ -0,0 +1,423 @@
+    silence_mollusk_logs();
+
+    let (pool, mut lsl, mut out_lsd, mut inp_lsd) = fixture_lst_state_data();
+
+    let withdraw_to = Pubkey::new_unique().to_bytes();
+    let builder = jupsol_wsol_builder(
+        pool.rebalance_authority,
+        out_lsd.lst_state.mint,
+        inp_lsd.lst_state.mint,
+        withdraw_to,
+    );
+
+    out_lsd.lst_state.sol_value_calculator = builder.out_calc_prog;
+    inp_lsd.lst_state.sol_value_calculator = builder.inp_calc_prog;
+
+    let out_idx = lsl.upsert(out_lsd) as u32;
+    let inp_idx = lsl.upsert(inp_lsd) as u32;
+
+    let owner = Pubkey::new_unique();
+    let owner_token_account = Pubkey::new_unique();
+
+    let inp_mint_account = mock_mint(raw_mint(None, None, 0, 9));
+    let inp_mint_decimals = RawMint::of_acc_data(&inp_mint_account.data)
+        .and_then(Mint::try_from_raw)
+        .expect("valid mint")
+        .decimals();
+
+    let inp_pool_reserves = Pubkey::new_from_array(inp_lsd.pool_reserves);
+
+    let preview_ixs = rebalance_ixs(&builder, out_idx, inp_idx, amount, 0, u64::MAX);
+
+    let transfer_ix = create_transfer_ix(
+        owner,
+        owner_token_account,
+        inp_lsd.lst_state.mint,
+        inp_mint_decimals,
+        inp_pool_reserves,
+        0,
+    );
+    let preview_chain = vec![preview_ixs[0].clone(), transfer_ix, preview_ixs[1].clone()];
+
+    let preview_accounts = setup_rebalance_transaction_accounts(
+        &builder,
+        &preview_chain,
+        &pool,
+        &lsl.lst_state_list,
+        &lsl.all_pool_reserves,
+        pool.rebalance_authority,
+        out_lsd.lst_state.mint,
+        inp_lsd.lst_state.mint,
+        withdraw_to,
+        amount * 2,
+        amount * 2,
+        owner,
+        owner_token_account,
+        0,
+    );
+
+    let start_result =
+        SVM.with(|svm| svm.process_instruction(&preview_chain[0], &preview_accounts));
+
+    let pool_pk = Pubkey::new_from_array(POOL_STATE_ID);
+    let rr_pk = Pubkey::new_from_array(REBALANCE_RECORD_ID);
+
+    let pool_after_start = start_result
+        .resulting_accounts
+        .iter()
+        .find(|(pk, _)| *pk == pool_pk)
+        .map(|(_, acc)| acc)
+        .expect("pool after start");
+
+    let rr_after_start = start_result
+        .resulting_accounts
+        .iter()
+        .find(|(pk, _)| *pk == rr_pk)
+        .map(|(_, acc)| acc)
+        .expect("rebalance record after start");
+
+    let pool_state_after = PoolStatePacked::of_acc_data(&pool_after_start.data)
+        .expect("pool state")
+        .into_pool_state();
+
+    let rebalance_record =
+        unsafe { RebalanceRecord::of_acc_data(&rr_after_start.data) }.expect("rebalance record");
+
+    // Calculate how many lamports the pool lost
+    let deficit = rebalance_record
+        .old_total_sol_value
+        .saturating_sub(pool_state_after.total_sol_value);
+
+    let owner_transfer_amount = deficit + (deficit / 100);
```

Comment:
is this `+ (deficit / 100)` just to bump the amount to make sure the threshold is always met? Alright man if its too hard for you to generalize across the different SolValueCalculator programs right now you can just test with `inp` always into `wsol` so that the amount is always 1:1 with sol_value